kore/src/Kore/Repl/Interpreter.hs

```diff
@@ -519,6 +520,34 @@ labelDel lbl = do
            putStrLn' "Removed label."
        else putStrLn' "Label doesn't exist."
 
+clear
+    :: forall level m
+    .  MonadState (ReplState level) m
+    => MonadWriter String m
+    => Maybe Int
+    -> m ()
+clear mi = do
+    node <- maybe <$> Lens.use lensNode <*> pure id <*> pure mi
+    eg@Strategy.ExecutionGraph { graph = gr } <- Lens.use lensGraph
+    let
+        nodes = collect (next gr) node
+        gr' = Graph.delNodes nodes gr
+        prevNode =
+            maybe 0 id
+                . listToMaybe
+                . fmap fst
+                $ Graph.lpre gr node
+    lensGraph .= eg { Strategy.graph = gr' }
+    lensNode .= prevNode
+    putStrLn' $ "Removed " <> show (length nodes) <> " node(s)."
+
+  where
+    next :: InnerGraph -> Graph.Node -> [Graph.Node]
+    next gr n = fst <$> Graph.lsuc gr n
+
+    collect :: (a -> [a]) -> a -> [a]
+    collect f x = x : [ z | y <- f x, z <- collect f y]
+
```

Comment:
If this function is applied on node 0 then the whole graph will be deleted and the `step`/`try` commands will throw an exception. Maybe we shouldn't let the user delete the initial node, since they already can initialize the proof for a different claim using the `prove` command. What do you think?